devtool build linux-raspberrypi fails

This was tested on current master. Only `poky/master` and `meta-raspberrypi/master` in `bblayers.conf`.

After starting a `devtool modify linux-raspberrypi` session, building the kernel fails with the following log. Any suggestion where to look for troubleshooting this?

```
MACHINE="raspberrypi3" devtool build linux-raspberrypi

Build Configuration:
BB_VERSION        = "1.33.1"
BUILD_SYS         = "x86_64-linux"
NATIVELSBSTRING   = "universal"
TARGET_SYS        = "arm-poky-linux-gnueabi"
MACHINE           = "raspberrypi3"
DISTRO            = "poky"
DISTRO_VERSION    = "2.2+snapshot-20170228"
TUNE_FEATURES     = "arm armv7ve vfp thumb neon vfpv4 callconvention-hard cortexa7"
TARGET_FPU        = "hard"
meta              
meta-poky         
meta-yocto-bsp    = "master:3c83b56309ab419f8cda72c0711479f60f61439a"
meta-raspberrypi  = "master:c768a3d1aed8d16d08afe12fddb359914e0a203a"
workspace         = "<unknown>:<unknown>"

Initialising tasks: 100% |###########################################################################################################################################################################| Time: 0:00:00
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
ERROR: linux-raspberrypi-1_4.4.48+git999-r0 do_configure: oe_runmake failed
ERROR: linux-raspberrypi-1_4.4.48+git999-r0 do_configure: Function failed: do_configure (log file is located at /scratch/gizero/poky-master-build/tmp/work/raspberrypi3-poky-linux-gnueabi/linux-raspberrypi/1_4.4.4
8+git999-r0/temp/log.do_configure.4531)
ERROR: Logfile of failure stored in: /scratch/gizero/poky-master-build/tmp/work/raspberrypi3-poky-linux-gnueabi/linux-raspberrypi/1_4.4.48+git999-r0/temp/log.do_configure.4531
Log data follows:
| DEBUG: Executing python function externalsrc_configure_prefunc
| DEBUG: Python function externalsrc_configure_prefunc finished
| DEBUG: Executing python function sysroot_cleansstate
| DEBUG: Python function sysroot_cleansstate finished
| DEBUG: Executing python function check_oldest_kernel
| DEBUG: Python function check_oldest_kernel finished
| DEBUG: Executing shell function do_configure
| NOTE: make HOSTCC=gcc  HOSTCPP=gcc  -E oldconfig
| ERROR: oe_runmake failed
| make: *** No rule to make target 'oldconfig'.  Stop.
| WARNING: exit code 1 from a shell command.
| ERROR: Function failed: do_configure (log file is located at /scratch/gizero/poky-master-build/tmp/work/raspberrypi3-poky-linux-gnueabi/linux-raspberrypi/1_4.4.48+git999-r0/temp/log.do_configure.4531)
ERROR: Task (/home/gizero/work/upstreaming/meta-raspberrypi/recipes-kernel/linux/linux-raspberrypi_4.4.bb:do_configure) failed with exit code '1'
NOTE: Tasks Summary: Attempted 452 tasks of which 128 didn't need to be rerun and 1 failed.
```

--- a/recipes-kernel/linux/linux-raspberrypi.inc
+++ b/recipes-kernel/linux/linux-raspberrypi.inc
@@ -152,8 +152,6 @@ do_configure_prepend() {
     # Remove all modified configs and add the rest to .config
     sed -e "${CONF_SED_SCRIPT}" < '${B}/.config.patched' >> '${B}/.config'
     rm -f ${B}/.config.patched
-
-    yes '' | oe_runmake oldconfig
 }
 
 do_compile_append_raspberrypi3-64() {

--- a/recipes-kernel/linux/linux-raspberrypi.inc
+++ b/recipes-kernel/linux/linux-raspberrypi.inc
@@ -61,7 +61,43 @@ kernel_configure_variable() {
     fi
 }
 
+config_setup() {
+    # From kernel.bbclass. Unfortunately, this is needed to support builds that
+    # use devtool. The reason is as follows:
+    #
+    # - In devtool builds, externalsrc.bbclass gets inherited and sets a list of
+    # SRCTREECOVEREDTASKS, which don't get run because they affect the source
+    # tree and, when using devtool, we want the developer's changes to be the
+    # single source of truth. kernel-yocto.bbclass adds do_kernel_configme to
+    # SRCTREECOVEREDTASKS, so it doesn't run in a devtool build., In a normal
+    # non-devtool build, do_kernel_configme creates ${B}.config.
+    #
+    # - Normally (e.g. in linux-yocto), it would be OK that do_kernel_configme
+    # doesn't run, because the first few lines of do_configure in kernel.bbclass
+    # populate ${B}.config from either ${S}.config (if it exists) for custom
+    # developer changes, or otherwise from ${WORDIR}/defconfig.
+    #
+    # - In linux-raspberrypi, we add do_configure_prepend, which tweaks
+    # ${B}.config. Since this runs *before* the kernel.bbclass do_configure,
+    # ${B}.config doesn't yet exist and we hit an error. Thus we need to move
+    # the logic from do_configure up to before our do_configure_prepend. Because
+    # we are copying only a portion of do_configure and not the whole thing,
+    # there is no clean way to do it using OE functionality, so we just
+    # copy-and-paste.
+    if [ "${S}" != "${B}" ] && [ -f "${S}/.config" ] && [ ! -f "${B}/.config" ]; then
+        mv "${S}/.config" "${B}/.config"
+    fi
+
+    # Copy defconfig to .config if .config does not exist. This allows
+    # recipes to manage the .config themselves in do_configure_prepend().
+    if [ -f "${WORKDIR}/defconfig" ] && [ ! -f "${B}/.config" ]; then
+        cp "${WORKDIR}/defconfig" "${B}/.config"
+    fi
+}
+
 do_configure_prepend() {
+    config_setup
+
     mv -f ${B}/.config ${B}/.config.patched
     CONF_SED_SCRIPT=""
 

